### PR TITLE
Improvements to library testing.

### PR DESCRIPTION
**Pull request by jmchilton**
- Add tests for creating a collection from a library.
- Add more class names to library GUI to improve selectors for tests.
- Automatically select whichever radio box makes sense in library collection builder dialog based on whether there is a selection or not.

Does this sound okay?

If this gets merged and broke tests (seems very possible) - I'll follow up with more fixes to your branch.

### CODE REVIEW

File: client/galaxy/scripts/mvc/library/library-foldertoolbar-view.js [hunk 1]
```diff
@@ -1382,12 +1382,12 @@ var FolderToolbarView = Backbone.View.extend({
                 "</label>",
                 "</div>",
                 // create new folder button
-                '<button style="display:none;" data-toggle="tooltip" data-placement="top" title="Create New Folder" class="btn btn-default primary-button toolbtn-create-folder add-library-items toolbar-item" type="button">',
+                '<button style="display:none;" data-toggle="tooltip" data-placement="top" title="Create New Folder" class="btn btn-default primary-button toolbtn-create-folder add-library-items add-library-items-folder toolbar-item" type="button">',
```
Comment (martenson) [inline client/galaxy/scripts/mvc/library/library-foldertoolbar-view.js:1385]: should we prefix classes used just for testing with something?

File: client/galaxy/scripts/mvc/library/library-foldertoolbar-view.js [hunk 2]
```diff
@@ -1828,13 +1828,13 @@ var FolderToolbarView = Backbone.View.extend({
                 "<h4>Which datasets?</h4>",
                 '<form class="form-inline">',
                 '<label class="radio-inline">',
-                '<input type="radio" name="radio_elements" id="selection_radio" value="selection" <% if (!selected_datasets) { %> disabled <% } %> > current selection',
+                '<input type="radio" name="radio_elements" id="selection_radio" value="selection" <% if (!selected_datasets) { %> disabled <% } else { %> checked <% } %> > current selection',
```
Comment (martenson) [inline client/galaxy/scripts/mvc/library/library-foldertoolbar-view.js:1831]: very useful, I wanted to add a validator for the radio selection but this is better

File: test/selenium_tests/test_library_to_collections.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
+from .framework import (
+    selenium_test,
+    SharedStateSeleniumTestCase,
+)
+
+
+class LibraryToCollectionsTestCase(SharedStateSeleniumTestCase):
+
+    requires_admin = True
```
Comment (martenson) [inline test/selenium_tests/test_library_to_collections.py:9]: but it shouldn't ?

Comment (jmchilton) [reply test/selenium_tests/test_library_to_collections.py:9]: All the Selenium library test cases do right now - ideally the test cases I guess would create libraries and assign their permissions to another user - but that is a bit more work.

File: client/galaxy/scripts/mvc/library/library-foldertoolbar-view.js (hunk 1 above)
Comment (jmchilton) [reply client/galaxy/scripts/mvc/library/library-foldertoolbar-view.js:1385]: From Gitter PM:

> I don't really consider them test-only CSS classes - they are useful for people customizing Galaxy's CSS, tour building, automated browser based tooling
The tool form prefixes labels a couple things with a tour_id attribute I think - I'm not a fan of that.
I'm okay with a prefix as long as it captures that this isn't specific to testing or tours etc...

**Comment by martenson** [comment]
Thanks John.

**Pull request merged**